File: app/src/main/java/top/easelink/lcg/ui/main/articles/viewmodel/FavoriteArticleItemViewModel.java
package top.easelink.lcg.ui.main.articles.viewmodel;

import androidx.databinding.ObservableField;
import org.greenrobot.eventbus.EventBus;
import top.easelink.lcg.ui.main.model.OpenArticleEvent;
import top.easelink.lcg.ui.main.source.model.ArticleEntity;

public class FavoriteArticleItemViewModel {

    public final ObservableField<String> title;
    public final ObservableField<String> author;
    public final ObservableField<String> content;


    private final ArticleEntity articleEntity;

    FavoriteArticleItemViewModel(ArticleEntity articleEntity) {
        this.articleEntity = articleEntity;
        title = new ObservableField<>(articleEntity.getTitle());
        author = new ObservableField<>(articleEntity.getAuthor());
        content = new ObservableField<>(articleEntity.getContent());
    }

    public void onItemClick() {
        OpenArticleEvent event = new OpenArticleEvent(articleEntity.getUrl());
        EventBus.getDefault().post(event);
    }
}

File: app/src/main/java/top/easelink/lcg/ui/main/articles/viewmodel/FavoriteArticleItemViewModel.kt
package top.easelink.lcg.ui.main.articles.viewmodel

import androidx.lifecycle.MutableLiveData
import org.greenrobot.eventbus.EventBus
import top.easelink.lcg.ui.main.model.OpenArticleEvent
import top.easelink.lcg.ui.main.source.model.ArticleEntity

class FavoriteArticleItemViewModel internal constructor(private val articleEntity: ArticleEntity) {

    val title: MutableLiveData<String> = MutableLiveData(articleEntity.title)
    val author: MutableLiveData<String> = MutableLiveData(articleEntity.author)
    val content: MutableLiveData<String> = MutableLiveData(articleEntity.content)

    fun onItemClick() {
        val event = OpenArticleEvent(articleEntity.url)
        EventBus.getDefault().post(event)
    }

}